csharp/GrpcOpenTracing/GrpcOpenTracing/GrpcTraceLogger.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using OpenTracing;
+
+namespace GrpcOpenTracing
+{
+    internal class GrpcTraceLogger<TRequest, TResponse> 
+        where TRequest : class 
+        where TResponse : class
+    {
+        private readonly ISpan span;
+
+        public GrpcTraceLogger(ISpan span)
+        {
+            this.span = span;
+        }
+
+        public void Request(TRequest req)
+        {
+            this.span.Log(new Dictionary<string, object>
+            {
+                { LogFields.Event, "gRPC request" },
+                { "data", req }
+            });
+        }
+
+        public void Response(TResponse rsp)
+        {
+            this.span.Log(new Dictionary<string, object>
+            {
+                { LogFields.Event, "gRPC response" },
+                { "data", rsp }
```

Comment:
Logging the whole response seems wrong. Would be best to compare this with the Java implementation and check what fields they log.

Reply:
Java is only logging the requests but not the response. I want to have it configurable and only send it if verbose is active.